pkg/crc/cluster/cluster.go

```diff
@@ -119,20 +119,56 @@ func UpdateClusterID(ocConfig oc.Config) error {
 	return nil
 }
 
-func AddProxyConfigToCluster(ocConfig oc.Config, proxy *network.ProxyConfig) error {
+func AddProxyConfigToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig) error {
+	trustedCAName := "user-ca-bundle"
 	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
 		fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}'`, proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString()),
 		"-n", "openshift-config", "--type", "merge"}
 
 	if err := WaitForOpenshiftResource(ocConfig, "proxy"); err != nil {
 		return err
 	}
+	if proxy.ProxyCACert != "" {
+		logging.Debug("Adding proxy CA cert to cluster")
+		if err := addProxyCACertToCluster(sshRunner, ocConfig, proxy, trustedCAName); err != nil {
+			return err
+		}
+		cmdArgs = []string{"patch", "proxy", "cluster", "-p",
+			fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s", "trustedCA":{"name":"%s"}}}'`,
+				proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString(), trustedCAName),
+			"-n", "openshift-config", "--type", "merge"}
```

Comment:
Let's not duplicate this. Maybe this is doable with a `struct` serialized to json rather than a `fmt.Sprintf` in order to handle the serialization of the optional `trustedCA` more easily?